mmgen/datasets/pipelines/augmentation.py

```diff
@@ -128,13 +128,17 @@ def __init__(self,
             assert size_factor is not None, (
                 'When max_size is used, '
                 f'size_factor should also be set. But received {size_factor}.')
-        if isinstance(scale, float):
+        if isinstance(scale, float, int):
+            assert keep_ratio, ('When scale is a single float/int,'
+                                ' keep_ratio must be ture')
             if scale <= 0:
                 raise ValueError(f'Invalid scale {scale}, must be positive.')
         elif mmcv.is_tuple_of(scale, int):
             max_long_edge = max(scale)
             max_short_edge = min(scale)
             if max_short_edge == -1:
+                assert keep_ratio, ('When scale includes a -1, '
```

Comment:
When -1 in the given `scale`, we manually calculate the size of image. Therefore we should use `mmcv.imresize` and `keep_ratio` should not be True. I wonder if this can pass the unit test.

Reply:
> When -1 in the given `scale`, we manually calculate the size of image. Therefore we should use `mmcv.imresize` and `keep_ratio` should not be True. I wonder if this can pass the unit test.

My bad, forget to do the test. But don't you think that here `keep_ratio` as true is logically correct, since the size is computed according to the ratio?

Reply:
Maybe setting `keep_ratio` as `True` is logically correct because we calculate the image size manually. However, in the current code, we would call `mmcv.imrescale` when `keep_ratio` is `True`. I suggest in this PR, we only fix the bug of size misplace and we can try to refactor the `Resize`  class later.

Reply:
I think you should convert `assert keep_ratio` to `assert not keep_ratio`.